### PR TITLE
Verify red-pajama-3b-chat with GenAI text_generation

### PR DESCRIPTION
**Pull request by tranchung163**
This pull request is focused on expanding the test coverage for the [redpajama-3b-chat ](https://huggingface.co/ikala/redpajama-3b-chat)Large Language Model (LLM) within OpenVINO GenAI. The redpajama-3b model is a significant addition to the supported models list, and testing its functionality. (Issue #263)

### CODE REVIEW

**Comment by tranchung163** [comment]
Hello @pavel-esir,
can you take a look at this PR ?

File: .github/workflows/causal_lm_cpp.yml [hunk 1]
```diff
@@ -351,3 +351,29 @@ jobs:
...
+          source ./ov/setupvars.sh
+          python -m pip install --upgrade-strategy eager "optimum>=1.14" -r ./llm_bench/python/requirements.txt ./thirdparty/openvino_tokenizers/[transformers] --extra-index-url https://download.pytorch.org/whl/cpu && python ./llm_bench/python/convert.py --model_id ikala/redpajama-3b-chat --output_dir ./redpajama-3b-chat/ --precision FP16 &
+          cmake -DCMAKE_BUILD_TYPE=Release -S ./text_generation/causal_lm/cpp/ -B ./build/
+          cmake --build ./build/ --config Release -j 15
+          wait
+      - name: Run Generation
+        run: |
+          source ./ov/setupvars.sh
+          convert_tokenizer ./redpajama-3b-chat/pytorch/dldt/FP16/ --output ./redpajama-3b-chat/pytorch/dldt/FP16/ --with-detokenizer --trust-remote-code
+          timeout 50s ./build/beam_search_causal_lm ./redpajama-3b-chat/pytorch/dldt/FP16/ 69 > ./pred.txt
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:379]: could you please also add comparison with Hugging Face?
Example is https://github.com/openvinotoolkit/openvino.genai/blob/f973f6287ed5d9decf1bd3e58e88e6c813e24378/.github/workflows/causal_lm_cpp.yml#L67-L80

Comment (tranchung163) [reply .github/workflows/causal_lm_cpp.yml:379]: Hi @ilya-lavrenov, thanks for reviewing. I have added comparison with Hugging Face (#320). Can you take a look ?

File: .github/workflows/causal_lm_cpp.yml [hunk 2]
```diff
@@ -376,4 +376,46 @@ jobs:
                   raise RuntimeError(f'Missing "{ref=}" from predictions')
               predictions = predictions[:idx] + predictions[idx + len(ref):]
           "
-          echo Phi-1_5 passed
\ No newline at end of file
+          echo Phi-1_5 passed
+  cpp-beam_search_causal_lm-redpajama-3b-chat:
+    runs-on: ubuntu-22.04-8-cores
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:381]:
```suggestion
    runs-on: ubuntu-20.04-8-cores
```

File: .github/workflows/causal_lm_cpp.yml [hunk 3]
```diff
@@ -376,4 +376,46 @@ jobs:
...
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.10
+      - name: Install OpenVINO
+        run: |
+          mkdir ./ov/
+          curl https://storage.openvinotoolkit.org/repositories/openvino/packages/2024.0/linux/l_openvino_toolkit_ubuntu22_2024.0.0.14509.34caeefd078_x86_64.tgz | tar --directory ./ov/ --strip-components 1 -xz
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:392]:
```suggestion
          curl https://storage.openvinotoolkit.org/repositories/openvino/packages/2024.0/linux/l_openvino_toolkit_ubuntu20_2024.0.0.14509.34caeefd078_x86_64.tgz | tar --directory ./ov/ --strip-components 1 -xz
```

File: .github/workflows/causal_lm_cpp.yml [hunk 4]
```diff
@@ -376,4 +376,46 @@ jobs:
...
+          sudo ./ov/install_dependencies/install_openvino_dependencies.sh
+      - name: Download, convert and build
+        run: |
+          source ./ov/setupvars.sh
+          python -m pip install --upgrade-strategy eager "optimum>=1.14" -r ./llm_bench/python/requirements.txt ./thirdparty/openvino_tokenizers/[transformers] --extra-index-url https://download.pytorch.org/whl/cpu && python ./llm_bench/python/convert.py --model_id ikala/redpajama-3b-chat --output_dir ./redpajama-3b-chat/ --precision FP16 &
+          cmake -DCMAKE_BUILD_TYPE=Release -S ./text_generation/causal_lm/cpp/ -B ./build/
+          cmake --build ./build/ --config Release -j 15
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:399]:
```suggestion
          cmake --build ./build/ --config Release -j
```

File: .github/workflows/causal_lm_cpp.yml [hunk 5]
```diff
@@ -376,4 +376,46 @@ jobs:
...
+          echo Phi-1_5 passed
+  cpp-beam_search_causal_lm-redpajama-3b-chat:
+    runs-on: ubuntu-20.04-8-cores
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.10
```
Comment (ilya-lavrenov) [inline .github/workflows/causal_lm_cpp.yml:388]:
```suggestion
          python-version: 3.8
```

**Comment by ilya-lavrenov** [comment]
Please, fix merge conflicts

File: .github/workflows/causal_lm_cpp.yml [hunk 6]
```diff
@@ -395,4 +395,47 @@ jobs:
                   raise RuntimeError(f'Missing "{ref=}" from predictions')
               predictions = predictions[:idx] + predictions[idx + len(ref):]
           "
+          echo Phi-1_5 passed
+          
+  cpp-greedy_causal_lm-redpajama-3b-chat:
+    runs-on: ubuntu-20.04-8-cores
```
Comment (Wovchena) [inline .github/workflows/causal_lm_cpp.yml:401]:
```suggestion
    runs-on: ubuntu-20.04
```
Can you try the smaller runner?

**Review by Wovchena — CHANGES_REQUESTED** [review]

**Review by Wovchena — APPROVED** [review]
Thank you!

**Pull request merged**